core/src/main/java/com/graphhopper/reader/dem/EdgeSampling.java

```diff
@@ -0,0 +1,72 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.reader.dem;
+
+import com.graphhopper.util.DistanceCalc;
+import com.graphhopper.util.DistanceCalcEarth;
+import com.graphhopper.util.PointList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Ensures that elevation is sampled along a point list with no more than maxDistance between samples. Works by adding
+ * points along long edges and fetching elevation at each inserted point.
+ *
+ * This uses a simple linear approximation to interpolate between points which should be fine for short distances, but
+ * if support for longer distances is needed or to handle segments crossing the international date line, should use a
+ * more robust algorithm that interpolates along the great circle path between points.
```

Comment:
Do we think that this assumption and approximation are fine? Or should we use a more exact calculation to interpolate between points.

Reply:
Yes I think this is a good decision. I did something like this using the great circle. This is quite computation heavy and probably not necessary in most cases.